File: Red.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Red
{
    public DataObject data { get; set; }
    public List<int> rn { get; set; }
    public int epochs { get; set; }
    public int checks { get; set; } = 4;
    public List<double> w { get; set; }
}

public class DataObject
{
    public List<List<double>> x { get; set; }
    public List<List<double>> y { get; set; }
}

File: Red.java
package org.example;

import java.util.List;

public class Red {
    public DataObject data;
    public List<Integer> rn;
    public int epochs;
    public int checks = 4;
    public List<Double> w;
    class DataObject {
        public List<List<Double>> x;
        public List<List<Double>> y;
    }
}

